sql/hive/src/main/scala/org/apache/spark/sql/hive/TableReader.scala

```diff
@@ -311,6 +309,36 @@ class HadoopTableReader(
     // Only take the value (skip the key) because Hive works only with values.
     rdd.map(_._2)
   }
+
+  /**
+   * If `spark.sql.hive.fileInputFormat.enabled` is true, this function will optimize the input
+   * method(including format and the size of splits) while reading Hive tables.
+   */
+  private def getAndOptimizeInput(
+    inputClassName: String): Class[InputFormat[Writable, Writable]] = {
+
+    var ifc = Utils.classForName(inputClassName)
+      .asInstanceOf[java.lang.Class[InputFormat[Writable, Writable]]]
+    if (conf.getConf(HiveUtils.HIVE_FILE_INPUT_FORMAT_ENABLED)) {
+      hadoopConf.set("mapreduce.input.fileinputformat.split.maxsize",
```

Comment:
sorry if I rushed to read. but isn't it possible to manually set via configurations? For instance, `spark.hadoop.mapreduce.input.fileinputformat.split.maxsize`

Reply:
Thank you for your review! We named it as `spark.sql.hive.fileInputFormat.split.maxsize` now and it can take effect by replacing `mapreduce.input.fileinputformat.split.maxsize` of hadoopConf. Because we are not able to change the configurations of hadoop when use SparkSQL.

Reply:
Have you tried to run the following SQL command: "SET mapreduce.input.fileinputformat.split.maxsize = 134217728"?

Reply:
Can you also try `spark.hadoop.mapreduce.input.fileinputformat.split.maxsize`? `spark.hadoop` prefix is the way to set the hadoop properties.

Reply:
oops.. tried before but now it works! It's my fault.. I'll remove the redundant code and change the `spark.sql.hive.fileInputFormat.enabled` to `spark.sql.hive.inputFormat.optimizer.enabled`.